### PR TITLE
set content type and no-cache headers to blob publishing

### PR DESCRIPTION
**Pull request by riarenas**
https://github.com/dotnet/core-eng/issues/15377

We lost the headers we used to set when moving to the new publisher for blobs, so badges are getting the wrong content type when being uploaded. 

I'm reusing the logic we previously had in place for this. Let me know if you'd rather I copy these functions to some other utility class. 

### To double check:

* [ ] The right tests are in and and the right validation has happened.  Guidance:  https://github.com/dotnet/core-eng/tree/master/Documentation/Validation

### CODE REVIEW

**Comment by riarenas** [comment]
I'll make sure to try a publishing job using an sdk with these changes if this is the way we want to go, have a test build here producing an sdk: https://dev.azure.com/dnceng/internal/_build/results?buildId=1565422&view=results

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/AzureStorageAssetPublisher.cs [hunk 1]
```diff
@@ -52,6 +53,7 @@ public async Task PublishAssetAsync(string file, string blobPath, PushOptions op
 
                 try
                 {
+                    await blobClient.SetHttpHeadersAsync(AzureStorageUtils.GetBlobHeadersByExtension(file));
```
Comment (alexperovich) [inline src/Microsoft.DotNet.Build.Tasks.Feed/src/AzureStorageAssetPublisher.cs:56]: I think these headers need to be specified in the Upload call, in the BlobUploadOptions.HttpHeaders property. The blob doesn't (usually) exist before it is uploaded, so this call will fail.

Comment (riarenas) [reply src/Microsoft.DotNet.Build.Tasks.Feed/src/AzureStorageAssetPublisher.cs]: Fixed I think.

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/AzureStorageAssetPublisher.cs [hunk 2]
```diff
@@ -52,7 +54,11 @@ public async Task PublishAssetAsync(string file, string blobPath, PushOptions op
 
                 try
                 {
-                    await blobClient.UploadAsync(file);
+                    BlobUploadOptions blobUploadOptions = new()
+                    {
+                        HttpHeaders = AzureStorageUtils.GetBlobHeadersByExtension(file)
```
Comment (alexperovich) [inline src/Microsoft.DotNet.Build.Tasks.Feed/src/AzureStorageAssetPublisher.cs:59]: This probably needs the cache options too.

Comment (riarenas) [reply src/Microsoft.DotNet.Build.Tasks.Feed/src/AzureStorageAssetPublisher.cs:59]: Are you referring to something different than what GetBlobHeadersByExtension sets up in LN 159?

Comment (alexperovich) [reply src/Microsoft.DotNet.Build.Tasks.Feed/src/AzureStorageAssetPublisher.cs:59]: O, yea that method does it. Carry on.

**Comment by riarenas** [comment]
I think I'll be able to verify once this publishing build completes: https://dev.azure.com/dnceng/internal/_build/results?buildId=1565830&view=logs&j=ba23343f-f710-5af9-782d-5bd26b102304. It should be using an sdk with these changes.

**Comment by riarenas** [comment]
It works! https://dotnetcli.azureedge.net/dotnet/Sdk/7.0.100-alpha.1.22071.3/win_x86_Release_version_badge.svg has the right content type set.

**Comment by riarenas** [comment]
I took an already existing installer build and published that to general-testing, I'm going to try this with a new build

**Comment by riarenas** [comment]
Yep that was the previous problem. Since I took a build that had already published blobs, they didn't get updated. I ran this new build of installer in a test branch: https://dev.azure.com/dnceng/internal/_build/results?buildId=1565904&view=results

and then published it here: https://dev.azure.com/dnceng/internal/_build/results?buildId=1566020&view=logs&j=ba23343f-f710-5af9-782d-5bd26b102304&t=c7a8693b-2f9c-5ea8-c909-cde9405ac2e1

https://dotnetbuilds.azureedge.net/public/Sdk/7.0.100-alpha.1.22071.7/linux-musl_arm64_Release_version_badge.svg loads just fine from dotnetbuilds and it has the right content type set. 


@alexperovich PTAL when you can.

**Review by alexperovich — APPROVED** [review]

**Pull request merged**